### PR TITLE
Isolate IpfsClient/KuboClient implementation from Gateway into `noosphere_ipfs`

### PR DESCRIPTION
**Pull request by jsantell**

### CODE REVIEW

File: rust/noosphere-cli/src/native/commands/serve/gateway.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use noosphere_api::route::Route as GatewayRoute;
 use noosphere_storage::NativeStorage;
 
 use crate::native::commands::serve::{
-    ipfs::start_ipfs_syndication,
+    ipfs_syndication::start_ipfs_syndication,
```
Comment (cdata) [inline rust/noosphere-cli/src/native/commands/serve/gateway.rs:19]: Let's just keep this `ipfs` unless we have an egregious name ambiguity

File: rust/noosphere-ipfs/src/client.rs [hunk 2]
```diff
@@ -25,4 +25,7 @@ pub trait IpfsClient {
     async fn syndicate_blocks<R>(&self, car: R) -> Result<()>
     where
         R: AsyncRead + Send + Sync + 'static;
+
+    /// Returns the associated blocks (referenced by [Cid]) if found.
```
Comment (cdata) [inline rust/noosphere-ipfs/src/client.rs:29]: nit: Should be `block` singular, as the return value is always one block.

File: rust/noosphere-ipfs/src/kubo.rs [hunk 3]
```diff
@@ -82,6 +82,23 @@ impl IpfsClient for KuboClient {
             other_status => Err(anyhow!("Unexpected status code: {}", other_status)),
         }
     }
+
+    async fn get(&self, _cid: &Cid) -> Result<Vec<u8>> {
+        /*
```
Comment (cdata) [inline rust/noosphere-ipfs/src/kubo.rs:87]: Remove comment?

File: rust/noosphere-ipfs/src/kubo.rs [hunk 4]
```diff
@@ -82,6 +82,23 @@ impl IpfsClient for KuboClient {
...
+        api_url.set_query(Some(&format!("arg={}", cid_base64)));
+
+        let request = Request::builder()
+            .method("POST")
+            .uri(&api_url.to_string())
+            .body(Body::empty())?;
+        let response = self.client.request(request).await?;
+        let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
+        */
+        Err(anyhow!("not implemented"))
```
Comment (cdata) [inline rust/noosphere-ipfs/src/kubo.rs:100]: Prefer `todo!("What is to be done")`

**Review by cdata — CHANGES_REQUESTED** [review]
This is awesome, thanks for doing this. There is also some red tape needed to integrate this with our Github Actions workflows for release. 

Some areas to look at: https://github.com/subconsciousnetwork/noosphere/blob/d741c8db401a3f491025b49c042c6055b1175f9f/.github/workflows/release.yaml#L26-L35

https://github.com/subconsciousnetwork/noosphere/blob/d741c8db401a3f491025b49c042c6055b1175f9f/release-please-config.json#L7-L21

**Comment by jsantell** [comment]
Addressed feedback, fixed wasm tests; unsure about the expected return type of `IpfsClient::get(Cid)` until digging into the http RPC more

File: rust/README.md [hunk 5]
```diff
@@ -67,7 +67,7 @@ If you wish to compile for targets other than your local platform target, you
 can use rustup to get other targets easily. For example:
 
 ```sh
-rustup target install wasm32-unknown-unknown
+rustup target add wasm32-unknown-unknown
```
Comment (jsantell) [inline rust/README.md:70]: Looks like rustup uses `add` now

**Review by cdata — APPROVED** [review]
Thanks for this refactor / cleanup!

File: rust/README.md (hunk 5 above)
Comment (cdata) [reply rust/README.md:70]: Probably my typo TBH

**Pull request merged**